src/unix/linux/x11/audio.c

```diff
@@ -54,9 +56,9 @@ MTY_Audio *MTY_AudioCreate(uint32_t sampleRate, uint32_t minBuffer, uint32_t max
 	snd_pcm_hw_params_any(ctx->pcm, params);
 
 	snd_pcm_hw_params_set_access(ctx->pcm, params, SND_PCM_ACCESS_RW_INTERLEAVED);
-	snd_pcm_hw_params_set_format(ctx->pcm, params, SND_PCM_FORMAT_S16);
-	snd_pcm_hw_params_set_channels(ctx->pcm, params, channels);
-	snd_pcm_hw_params_set_rate(ctx->pcm, params, sampleRate, 0);
+	snd_pcm_hw_params_set_format(ctx->pcm, params, format->sampleFormat == MTY_AUDIO_SAMPLE_FORMAT_FLOAT ? SND_PCM_FORMAT_FLOAT : SND_PCM_FORMAT_S16);
+	snd_pcm_hw_params_set_channels(ctx->pcm, params, format->channels);
+	snd_pcm_hw_params_set_rate(ctx->pcm, params, format->sampleRate, 0);
 	snd_pcm_hw_params(ctx->pcm, params);
 	snd_pcm_nonblock(ctx->pcm, 1);
```

Comment:
Missing a channel map;

https://www.alsa-project.org/alsa-doc/alsa-lib/group___p_c_m.html#ga60ee7d2c2555e21dbc844a1b73839085
https://gist.github.com/raydudu/5590a196b9446c709c58a03eff1f38bc

Reply:
Not particularly tricky, but the `channelsMask` is passed as an opaque `uint32_t` because that's what windows and macos want. Turns out that Android uses that as well. But alsa on linux wants an explicit channel map.

I'll add a note for now with your link and defer this to another PR.